tests/testthat/test-license_note.R

```diff
@@ -7,6 +7,4 @@ test_that("LICENSE.note is generated properly", {
   write_license_note()
 
   expect_snapshot(cat_file("LICENSE.note"))
-  expect_rextendr_error(write_license_note(), NA)
-  expect_rextendr_error(write_license_note(force = FALSE), "LICENSE.note already exists.")
```

Comment:
Why were these tests removed?

Reply:
A couple of things:
- [`write_file()`](https://github.com/extendr/rextendr/blob/main/R/write_file.R) has an `overwrite` parameter. I updated the parameter name in `write_license_note()` from `force` to `overwrite` to make it consistent.
- `write_file()` also has its own check to see if the file exists, but it only emits a warning and returns NULL rather than an error. I figured this check only really needs to be handled in one place, so I removed it from `write_license_note()`.  

That means the test can't pass, so I was thinking I could either re-write it or update `write_file()` to emit an error. But for now, I just removed those tests.

Does this count as mission creep?

Reply:
yes it definitely does! Let's create a new issue and add it to the 0.4.0 milestone